pallets/transaction-multi-payment/src/lib.rs

```diff
@@ -341,6 +355,139 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		// Dispatch EVM permit
+		#[pallet::call_index(4)]
+		#[pallet::weight(
+			<T as Config>::EvmPermit::dispatch_weight(*gas_limit)
+		)]
+		pub fn dispatch_permit(
+			origin: OriginFor<T>,
+			currency: AssetIdOf<T>,
```

Comment:
thinking more about it this lets dispatcher to change payment currency without your approval. best would be to require user to call set_currency (though permit) 1st to change it same as we do with regular set_currency. regular evm call interaction with RPC (not permit) would always charge WETH. ofc the evm calls incoming from extrinsics would behave the same as they cuirrenctly are to enable substrate users to pay in any currency